src/flow/domain/mod.rs

```diff
@@ -593,8 +607,19 @@ impl Domain {
         }
     }
 
-    fn seed_replay(&mut self, tag: Tag, key: &[DataType]) {
-        let (m, source) = match self.replay_paths[&tag] {
+    fn seed_replay(&mut self, tag: Tag, key: &[DataType], transaction_state: Option<ReplayTransactionState>) {
+        if transaction_state.is_none() {
```

Comment:
Also, this seems to be the only place where `schedule_replay` is called? Shouldn't it also be the case for when you are a `TriggerEndpoint::Start` (i.e., when you're going to send a replay to someone else). Those presumably also need timestamps?